src/directives/layertree.js

```diff
@@ -220,6 +214,9 @@ ngeo.LayertreeController = function($scope, $element, $attrs) {
 
   $scope['layertreeCtrl'] = this;
 
+  $scope.$on('$destroy', angular.bind(this, function() {
+    this.map.removeLayer(this.layer);
```

Comment:
I don't like to do this in ngeo but that's the cleaner (only cleaner) manner to remove layer when this tree 'on destroy'.

The only other way to do this is to proceed like the "ngeo-layertree-nodelayer". this attribute refers to a fonction (that help to link a layer to a node (a layertree).
I can do a similar system with a "ngeo-layertree-onevents" attributes that allow to bind custom function to layertree events. In that function, we can probably link a 'on destroy function'.

(On writing that, I just convince myself that's a better idea. I'll try this. Wait...)